# PROYECTO FINAL 

##### Ejecutamos tecnicas de ETL de 4 dataframe que se usaran para completar el análisis sobre los jugadores que contienen los datos específicos de la temporada (2021-2022-2023) de cada jugador (puntos, bloqueos, asistencias, salarios, entre otros).

Se importan las librerias que se consideraron para realizar el ETL en esta fase del proyecto

In [1]:
import pandas as pd
import numpy as np

ANALISIS DE LA TEMPORADA 2021

Carga del dataframe "nba2021PerGame"

Revision del dataframe (info, conteo y nulos)

In [2]:
df1 = pd.read_csv("nba2021_per_game.csv") 

In [3]:
# Primeras filas del dataframe
print(df1.head(11))

                      Player Pos  Age   Tm   G  GS    MP   FG   FGA    FG%  \
0           Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590   
1               Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125   
2               Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603   
3                Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573   
4          LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476   
5          Ty-Shon Alexander  SG   22  PHO   3   0   2.7  0.0   1.0  0.000   
6   Nickeil Alexander-Walker  SG   22  NOP  23   3  19.2  3.3   8.2  0.410   
7              Grayson Allen  SG   25  MEM  19   8  23.9  3.2   7.4  0.429   
8              Jarrett Allen   C   22  TOT  28  10  26.2  4.4   6.8  0.642   
9              Jarrett Allen   C   22  BRK  12   5  26.7  3.7   5.4  0.677   
10             Jarrett Allen   C   22  CLE  16   5  25.9  4.9   7.8  0.624   

    ...    FT%  ORB  DRB   TRB  AST  STL  BLK  TOV   PF   PTS  

In [4]:
# Información general sobre el dataframe
print(df1.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 497 entries, 0 to 496
Data columns (total 29 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Player  497 non-null    object 
 1   Pos     497 non-null    object 
 2   Age     497 non-null    int64  
 3   Tm      497 non-null    object 
 4   G       497 non-null    int64  
 5   GS      497 non-null    int64  
 6   MP      497 non-null    float64
 7   FG      497 non-null    float64
 8   FGA     497 non-null    float64
 9   FG%     497 non-null    float64
 10  3P      497 non-null    float64
 11  3PA     497 non-null    float64
 12  3P%     497 non-null    float64
 13  2P      497 non-null    float64
 14  2PA     497 non-null    float64
 15  2P%     497 non-null    float64
 16  eFG%    497 non-null    float64
 17  FT      497 non-null    float64
 18  FTA     497 non-null    float64
 19  FT%     497 non-null    float64
 20  ORB     497 non-null    float64
 21  DRB     497 non-null    float64
 22  TR

In [5]:
np.shape(df1)

(497, 29)

In [6]:
#Revision Nulos 
ValoresNull1 = df1.isnull().sum() [df1.isnull().sum()>0]
print(ValoresNull1)

Series([], dtype: int64)


CAMBIAR NOMBRE DE COLUMNAS 

Se realiza modificacion de nombres de columnas para unificar criterios con el resto de las tablas 

In [7]:
nuevos_nombres = {
    'Tm': 'Team',
    'G': 'GP'
}

#Renombrar las columnas especificadas
df1.rename(columns=nuevos_nombres, inplace=True)

print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  ...  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  ...   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  ...   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  ...   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  ...   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  ...   

     FT%  ORB  DRB  TRB  AST  STL  BLK  TOV   PF   PTS  
0  0.561  1.3  2.7  4.0  0.6  0.4  0.5  1.0  1.9   6.5  
1  0.000  0.0  0.5  0.5  0.3  0.0  0.0  0.0  0.2   0.3  
2  0.468  4.3  4.6  8.9  2.1  1.0  0.6  1.7  1.9   8.0  
3  0.841  1.9  7.3  9.2  5.3  1.0  1.0  3.0  2.6  19.9  
4  0.762  0.8  3.5  4.3  1.9  0.4  0.9  0.9  1.5  14.1  

[5 rows x 29 columns]


In [8]:
# Año que deseas agregar
Temporada = 2021

# Agregar una nueva columna 'Temporada' con el valor del año especificado
df1['Temp'] = Temporada

# Mostrar las primeras filas del DataFrame para verificar
print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  ...  ORB  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  ...  1.3   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  ...  0.0   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  ...  4.3   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  ...  1.9   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  ...  0.8   

   DRB  TRB  AST  STL  BLK  TOV   PF   PTS  Temp  
0  2.7  4.0  0.6  0.4  0.5  1.0  1.9   6.5  2021  
1  0.5  0.5  0.3  0.0  0.0  0.0  0.2   0.3  2021  
2  4.6  8.9  2.1  1.0  0.6  1.7  1.9   8.0  2021  
3  7.3  9.2  5.3  1.0  1.0  3.0  2.6  19.9  2021  
4  3.5  4.3  1.9  0.4  0.9  0.9  1.5  14.1  2021  

[5 rows x 30 columns]


In [9]:
print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  ...  ORB  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  ...  1.3   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  ...  0.0   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  ...  4.3   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  ...  1.9   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  ...  0.8   

   DRB  TRB  AST  STL  BLK  TOV   PF   PTS  Temp  
0  2.7  4.0  0.6  0.4  0.5  1.0  1.9   6.5  2021  
1  0.5  0.5  0.3  0.0  0.0  0.0  0.2   0.3  2021  
2  4.6  8.9  2.1  1.0  0.6  1.7  1.9   8.0  2021  
3  7.3  9.2  5.3  1.0  1.0  3.0  2.6  19.9  2021  
4  3.5  4.3  1.9  0.4  0.9  0.9  1.5  14.1  2021  

[5 rows x 30 columns]


##### A continuación duplicaremos la columna que contienen las abrevituras de los equipos, esto es porque hay un valor posible el cual se identifica con TOT, hace referencia a los momentos en que jugadores tuvieron mas de un equipo en una temporada. Por ende en un futuro se tomara el valor TOT como referencia para el analisis, entonces en una columna quedara identificado con TOT, y en la nueva columna TOTAL TEAM, el total de equipos en el que jugo en la temporada. Esto se hará mas adelante cuando esten todas las temporadas unificadas

In [10]:
# Duplicar la columna 'Tm' y renombrarla a 'Total Team'
df1['TOTAL TEAM'] = df1['Team']

# Mostrar las primeras filas del DataFrame para verificar
print(df1.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  ...  DRB  \
0   Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590  ...  2.7   
1       Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  ...  0.5   
2       Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603  ...  4.6   
3        Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  ...  7.3   
4  LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476  ...  3.5   

   TRB  AST  STL  BLK  TOV   PF   PTS  Temp  TOTAL TEAM  
0  4.0  0.6  0.4  0.5  1.0  1.9   6.5  2021         MIA  
1  0.5  0.3  0.0  0.0  0.0  0.2   0.3  2021         MIL  
2  8.9  2.1  1.0  0.6  1.7  1.9   8.0  2021         NOP  
3  9.2  5.3  1.0  1.0  3.0  2.6  19.9  2021         MIA  
4  4.3  1.9  0.4  0.9  0.9  1.5  14.1  2021         SAS  

[5 rows x 31 columns]


In [11]:
np.shape(df1)

(497, 31)

Ya tenemos limpio el dataset de la temporada 2020-2021. Lo guardaremos para más adelante, cuando hagamos el merge de las temporadas.

In [12]:
df1.to_csv("Temporada2020_2021.csv",sep=",")

# ANALISIS TEMPORADA 2021 - 2022

In [13]:
# importación de los datos.
df2 = pd.read_csv("NBAPlayerStatsRegular.csv", encoding='latin-1', delimiter=';') 

In [14]:
print(df2.head(11))

    Rk                    Player Pos  Age   Tm   G  GS    MP   FG   FGA  ...  \
0    1          Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  ...   
1    2              Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  ...   
2    3               Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  ...   
3    4              Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  ...   
4    5         LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  ...   
5    6  Nickeil Alexander-Walker  SG   23  TOT  65  21  22.6  3.9  10.5  ...   
6    6  Nickeil Alexander-Walker  SG   23  NOP  50  19  26.3  4.7  12.6  ...   
7    6  Nickeil Alexander-Walker  SG   23  UTA  15   2   9.9  1.1   3.2  ...   
8    7             Grayson Allen  SG   26  MIL  66  61  27.3  3.9   8.6  ...   
9    8             Jarrett Allen   C   23  CLE  56  56  32.3  6.6   9.7  ...   
10   9             Jose Alvarado  PG   23  NOP  54   1  15.4  2.4   5.4  ...   

      FT%  ORB  DRB   TRB  AST  STL  BL

In [15]:
# Información general sobre el dataframe
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 812 entries, 0 to 811
Data columns (total 30 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Rk      812 non-null    int64  
 1   Player  812 non-null    object 
 2   Pos     812 non-null    object 
 3   Age     812 non-null    int64  
 4   Tm      812 non-null    object 
 5   G       812 non-null    int64  
 6   GS      812 non-null    int64  
 7   MP      812 non-null    float64
 8   FG      812 non-null    float64
 9   FGA     812 non-null    float64
 10  FG%     812 non-null    float64
 11  3P      812 non-null    float64
 12  3PA     812 non-null    float64
 13  3P%     812 non-null    float64
 14  2P      812 non-null    float64
 15  2PA     812 non-null    float64
 16  2P%     812 non-null    float64
 17  eFG%    812 non-null    float64
 18  FT      812 non-null    float64
 19  FTA     812 non-null    float64
 20  FT%     812 non-null    float64
 21  ORB     812 non-null    float64
 22  DR

In [16]:
np.shape(df2)

(812, 30)

In [17]:
# consulta de valores nulos
ValoresNull2 = df2.isnull().sum() [df2.isnull().sum()>0]
print(ValoresNull2)

Series([], dtype: int64)


Revisando cantidad de veces que se repite un nombre.

In [18]:
conteo_unicos = df2["Player"].value_counts()
print(conteo_unicos)

Player
Greg Monroe            5
Danuel House Jr.       4
Wenyen Gabriel         4
James Ennis III        4
Juancho Hernangómez    4
                      ..
Eric Bledsoe           1
Trae Young             1
Omer Yurtseven         1
Cody Zeller            1
Precious Achiuwa       1
Name: count, Length: 605, dtype: int64


In [19]:
# Año que deseas agregar
Temporada2 = 2022

# Agregar una nueva columna 'Temporada' con el valor del año especificado
df2['Temp'] = Temporada2

# Mostrar las primeras filas del DataFrame para verificar
print(df2.head())

   Rk             Player Pos  Age   Tm   G  GS    MP   FG   FGA  ...  ORB  \
0   1   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  ...  2.0   
1   2       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  ...  4.6   
2   3        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  ...  2.4   
3   4       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  ...  1.0   
4   5  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  ...  1.6   

   DRB   TRB  AST  STL  BLK  TOV   PF   PTS  Temp  
0  4.5   6.5  1.1  0.5  0.6  1.2  2.1   9.1  2022  
1  5.4  10.0  3.4  0.9  0.8  1.5  2.0   6.9  2022  
2  7.6  10.1  3.4  1.4  0.8  2.6  3.1  19.1  2022  
3  1.7   2.7  0.7  0.2  0.3  0.5  1.1   4.1  2022  
4  3.9   5.5  0.9  0.3  1.0  0.9  1.7  12.9  2022  

[5 rows x 31 columns]


Aquí se filtra por los jugadores que tienen TOT en sus equipos de la temporada.

In [20]:
# Crear la columna TOTAL TEAMS concatenando los nombres de los equipos
df2['TOTAL TEAM'] = df2.groupby(['Player'])['Tm'].transform(lambda x: '/'.join(x[x != 'TOT']))

# Filtrar solo las filas donde el equipo es 'TOT'
df_tot = df2[df2['Tm'] == 'TOT'].copy()

pd.options.display.max_columns = 60
pd.options.display.max_columns

# Ver el resultado
df_tot.head(20)

Rk                    Player    Pos  Age   Tm   G  GS    MP   FG   FGA  \
5      6  Nickeil Alexander-Walker     SG   23  TOT  65  21  22.6  3.9  10.5   
11    10           Justin Anderson     SF   28  TOT  16   6  19.8  2.3   5.9   
22    19             D.J. Augustin     PG   34  TOT  55   2  16.1  1.7   4.0   
29    24         Marvin Bagley III     PF   22  TOT  48  25  23.9  4.7   9.3   
51    44           DeAndre' Bembry     SF   27  TOT  56  20  18.3  2.2   3.9   
54    45             D?vis Bert?ns     PF   29  TOT  56   0  14.4  1.8   4.9   
81    70             Armoni Brooks  SG-PG   23  TOT  54  11  15.6  1.8   5.4   
86    73         Charlie Brown Jr.     SG   24  TOT  22   2   8.0  0.5   1.8   
89    74        Chaundee Brown Jr.  SG-SF   23  TOT   5   2  20.8  2.0   6.4   
94    77               Moses Brown      C   22  TOT  40   6   8.6  1.6   2.7   
113   94          Vernon Carey Jr.      C   20  TOT   7   1   6.3  1.0   1.9   
116   95              Jevon Carter     PG   26  TOT  66   3  13.7  1.5   3.8   
121   98       Willie Cauley-Stein      C   28  TOT  20   2   9.2  0.8   1.8   
143  118          DeMarcus Cousins      C   31  TOT  48   7  15.0  3.1   6.8   
146  119          Robert Covington  PF-SF   31  TOT  71  42  27.3  2.9   7.0   
149  120              Torrey Craig     SF   31  TOT  78  16  20.5  2.6   5.7   
156  125                Seth Curry     SG   31  TOT  64  64  33.4  5.6  11.5   
172  139         Spencer Dinwiddie     PG   28  TOT  67  51  29.6  4.6  11.0   
175  140          Donte DiVincenzo     SG   25  TOT  42   1  24.0  2.8   8.0   
184  147               Jeff Dowtin     PG   24  TOT   9   0  12.0  0.9   2.9   

       FG%   3P  3PA    3P%   2P  2PA    2P%   eFG%   FT  FTA    FT%  ORB  \
5    0.372  1.6  5.2  0.311  2.3  5.3  0.433  0.449  1.2  1.7  0.743  0.6   
11   0.379  0.9  3.7  0.254  1.3  2.3  0.583  0.458  0.9  1.2  0.789  0.3   
22   0.423  1.2  3.0  0.415  0.5  1.1  0.448  0.577  0.7  0.8  0.884  0.2   
29   0.504  0.5  2.0  0.237  4.2  7.3  0.579  0.530  1.4  2.1  0.663  2.2   
51   0.561  0.3  0.6  0.417  1.9  3.3  0.589  0.595  0.4  0.6  0.600  0.8   
54   0.360  1.4  4.1  0.335  0.4  0.8  0.489  0.500  0.6  0.7  0.900  0.2   
81   0.339  1.4  4.6  0.297  0.4  0.7  0.600  0.467  0.3  0.4  0.857  0.4   
86   0.256  0.0  0.5  0.083  0.4  1.2  0.333  0.269  0.4  0.5  0.900  0.6   
89   0.313  1.2  3.6  0.333  0.8  2.8  0.286  0.406  1.0  1.2  0.833  0.2   
94   0.593  0.0  0.0  0.000  1.6  2.7  0.593  0.593  1.1  1.8  0.597  1.0   
113  0.538  0.0  0.0  0.000  1.0  1.9  0.538  0.538  0.9  1.9  0.462  1.1   
116  0.387  1.0  2.6  0.388  0.5  1.3  0.386  0.518  0.2  0.3  0.833  0.2   
121  0.457  0.1  0.1  0.500  0.8  1.7  0.455  0.471  0.1  0.1  0.500  0.7   
143  0.460  0.8  2.5  0.303  2.3  4.2  0.554  0.517  2.0  2.6  0.760  1.5   
146  0.420  1.8  4.8  0.378  1.1  2.2  0.509  0.548  0.8  1.0  0.841  0.8   
149  0.454  1.0  3.0  0.329  1.6  2.7  0.594  0.541  0.5  0.7  0.750  1.1   
156  0.487  2.5  5.8  0.422  3.2  5.7  0.553  0.594  1.3  1.5  0.872  0.3   
172  0.416  1.7  4.9  0.339  2.9  6.0  0.479  0.493  2.9  3.8  0.774  0.7   
175  0.351  1.8  5.2  0.339  1.0  2.8  0.373  0.461  1.7  2.0  0.843  1.0   
184  0.308  0.1  1.0  0.111  0.8  1.9  0.412  0.327  0.2  0.2  1.000  0.9   

     DRB  TRB  AST  STL  BLK  TOV   PF   PTS  Temp   TOTAL TEAM  
5    2.3  2.9  2.4  0.7  0.4  1.4  1.6  10.6  2022      NOP/UTA  
11   2.6  2.9  2.1  0.5  0.4  0.5  1.4   6.4  2022      CLE/IND  
22   1.0  1.2  1.9  0.3  0.0  1.0  0.7   5.3  2022      HOU/LAL  
29   4.9  7.0  0.8  0.5  0.4  0.8  1.9  11.3  2022      SAC/DET  
51   2.2  2.9  1.2  0.9  0.4  0.6  2.0   5.1  2022      BRK/MIL  
54   1.9  2.1  0.6  0.3  0.2  0.3  1.5   5.6  2022      WAS/DAL  
81   1.5  1.9  1.1  0.5  0.2  0.5  1.4   5.3  2022      HOU/TOR  
86   0.8  1.5  0.3  0.5  0.2  0.2  0.8   1.4  2022      DAL/PHI  
89   3.0  3.2  0.8  0.4  0.0  0.2  1.4   6.2  2022      LAL/ATL  
94   2.4  3.4  0.0  0.2  0.4  0.5  1.3   4.3  2

In [21]:
# Identificar los jugadores que han jugado en más de un equipo en una temporada
multi_team_players = df2.groupby(['Player'])['Tm'].transform('nunique') > 1

# Filtrar para quedarte solo con la fila "TOT" para los jugadores con múltiples equipos
df_filtered = df2[~multi_team_players | (df2['Tm'] == 'TOT')]
np.shape(df_filtered)


(605, 32)

Ahora notamos que ya quedo resuelto el tema de la columna TOTAL TEAM

In [22]:
df_filtered.head(20)

Rk                    Player Pos  Age   Tm   G  GS    MP    FG   FGA  \
0    1          Precious Achiuwa   C   22  TOR  73  28  23.6   3.6   8.3   
1    2              Steven Adams   C   28  MEM  76  75  26.3   2.8   5.1   
2    3               Bam Adebayo   C   24  MIA  56  56  32.6   7.3  13.0   
3    4              Santi Aldama  PF   21  MEM  32   0  11.3   1.7   4.1   
4    5         LaMarcus Aldridge   C   36  BRK  47  12  22.3   5.4   9.7   
5    6  Nickeil Alexander-Walker  SG   23  TOT  65  21  22.6   3.9  10.5   
8    7             Grayson Allen  SG   26  MIL  66  61  27.3   3.9   8.6   
9    8             Jarrett Allen   C   23  CLE  56  56  32.3   6.6   9.7   
10   9             Jose Alvarado  PG   23  NOP  54   1  15.4   2.4   5.4   
11  10           Justin Anderson  SF   28  TOT  16   6  19.8   2.3   5.9   
14  11             Kyle Anderson  PF   28  MEM  69  11  21.5   3.0   6.8   
15  12     Giannis Antetokounmpo  PF   27  MIL  67  67  32.9  10.3  18.6   
16  13    Thanasis Antetokounmpo  SF   29  MIL  48   6   9.9   1.5   2.7   
17  14           Carmelo Anthony  PF   37  LAL  69   3  26.0   4.6  10.5   
18  15              Cole Anthony  PG   21  ORL  65  65  31.7   5.5  14.0   
19  16                OG Anunoby  SF   24  TOR  48  48  36.0   6.4  14.5   
20  17          Ryan Arcidiacono  PG   27  NYK  10   0   7.6   0.6   1.2   
21  18              Trevor Ariza  SF   36  LAL  24  11  19.3   1.4   4.1   
22  19             D.J. Augustin  PG   34  TOT  55   2  16.1   1.7   4.0   
25  20               Deni Avdija  SF   21  WAS  82   8  24.2   3.0   7.1   

      FG%   3P  3PA    3P%   2P   2PA    2P%   eFG%   FT   FTA    FT%  ORB  \
0   0.439  0.8  2.1  0.359  2.9   6.1  0.468  0.486  1.1   1.8  0.595  2.0   
1   0.547  0.0  0.0  0.000  2.8   5.0  0.548  0.547  1.4   2.6  0.543  4.6   
2   0.557  0.0  0.1  0.000  7.3  12.9  0.562  0.557  4.6   6.1  0.753  2.4   
3   0.402  0.2  1.5  0.125  1.5   2.6  0.560  0.424  0.6   1.0  0.625  1.0   
4   0.550  0.3  1.0  0.304  5.1   8.8  0.578  0.566  1.9   2.2  0.873  1.6   
5   0.372  1.6  5.2  0.311  2.3   5.3  0.433  0.449  1.2   1.7  0.743  0.6   
8   0.448  2.4  5.9  0.409  1.5   2.7  0.533  0.588  1.0   1.1  0.865  0.5   
9   0.677  0.0  0.2  0.100  6.6   9.6  0.688  0.678  2.9   4.2  0.708  3.4   
10  0.446  0.6  2.0  0.291  1.8   3.4  0.538  0.500  0.7   1.0  0.679  0.5   
11  0.379  0.9  3.7  0.254  1.3   2.3  0.583  0.458  0.9   1.2  0.789  0.3   
14  0.446  0.5  1.6  0.330  2.5   5.2  0.481  0.484  1.0   1.5  0.638  1.0   
15  0.553  1.1  3.6  0.293  9.2  15.0  0.616  0.582  8.3  11.4  0.722  2.0   
16  0.547  0.0  0.3  0.143  1.4   2.4  0.596  0.555  0.6   1.0  0.630  0.8   
17  0.441  2.2  5.8  0.375  2.5   4.7  0.521  0.544  1.9   2.3  0.830  0.9   
18  0.391  2.0  6.0  0.338  3.5   8.0  0.432  0.464  3.3   3.9  0.854  0.5   
19  0.443  2.4  6.6  0.363  4.0   7.9  0.511  0.526  1.9   2.5  0.754  1.5   
20  0.500  0.4  0.9  0.444  0.2   0.3  0.667  0.667  0.0   0.0  0.000  0.0   
21  0.333  0.8  3.1  0.270  0.5   1.0  0.520  0.434  0.4   0.8  0.556  0.4   
22  0.423  1.2  3.0  0.415  0.5   1.1  0.448  0.577  0.7   0.8  0.884  0.2   
25  0.432  1.0  3.2  0.317  2.0   3.9  0.525  0.503  1.3   1.7  0.757  0.6   

    DRB   TRB  AST  STL  BLK  TOV   PF   PTS  Temp TOTAL TEAM  
0   4.5   6.5  1.1  0.5  0.6  1.2  2.1   9.1  2022        TOR  
1   5.4  10.0  3.4  0.9  0.8  1.5  2.0   6.9  2022        MEM  
2   7.6  10.1  3.4  1.4  0.8  2.6  3.1  19.1  2022        MIA  
3   1.7   2.7  0.7  0.2  0.3  0.5  1.1   4.1  2022        MEM  
4   3.9   5.5  0.9  0.3  1.0  0.9  1.7  12.9  2022        BRK  
5   2.3   2.9  2.4  0.7  0.4  1.4  1.6  10.6  2022    NOP/UTA  
8   2.9   3.4  1.5  0.7  0.3  0.7  1.5  11.1  2022        MIL  
9   7.3  10.8  1.6  0.8  1.3  1.7  1.7  16.1  2022        CLE  
10  1.4   1.9  2.8  1.3  0.1  0.7  1.4   6.1  2022        NOP  
11  2.6   2.9  2.1  0.5  0.4  0.5  1.4   6.4  2022    CLE/IND  
14  4.3   5.3  2.7  1.1  0.7  1.0  1.6   7.6  2022        MEM  


Consultamos si ahora los nombres se repiten

In [23]:
conteo_unicos = df_filtered["Player"].value_counts()
print(conteo_unicos)

Player
Ivica Zubac         1
Precious Achiuwa    1
Steven Adams        1
Bam Adebayo         1
Santi Aldama        1
                   ..
Trevor Ariza        1
Ryan Arcidiacono    1
OG Anunoby          1
Cole Anthony        1
Carmelo Anthony     1
Name: count, Length: 605, dtype: int64


In [24]:
np.shape(df_filtered)

(605, 32)

Dropeamos la columna de los indices, porque son incorrectos.

In [25]:
df_filtered = df_filtered.drop(columns=['Rk'])

In [26]:
nuevos_nombres = {
    'Tm': 'Team',
    'G': 'GP'
}

# Renombrar las columnas especificadas
df_filtered.rename(columns=nuevos_nombres, inplace=True)


# Mostrar las primeras filas del DataFrame para verificar los cambios
print(df_filtered.head())

              Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
0   Precious Achiuwa   C   22  TOR  73  28  23.6  3.6   8.3  0.439  0.8  2.1   
1       Steven Adams   C   28  MEM  76  75  26.3  2.8   5.1  0.547  0.0  0.0   
2        Bam Adebayo   C   24  MIA  56  56  32.6  7.3  13.0  0.557  0.0  0.1   
3       Santi Aldama  PF   21  MEM  32   0  11.3  1.7   4.1  0.402  0.2  1.5   
4  LaMarcus Aldridge   C   36  BRK  47  12  22.3  5.4   9.7  0.550  0.3  1.0   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  AST  STL  \
0  0.359  2.9   6.1  0.468  0.486  1.1  1.8  0.595  2.0  4.5   6.5  1.1  0.5   
1  0.000  2.8   5.0  0.548  0.547  1.4  2.6  0.543  4.6  5.4  10.0  3.4  0.9   
2  0.000  7.3  12.9  0.562  0.557  4.6  6.1  0.753  2.4  7.6  10.1  3.4  1.4   
3  0.125  1.5   2.6  0.560  0.424  0.6  1.0  0.625  1.0  1.7   2.7  0.7  0.2   
4  0.304  5.1   8.8  0.578  0.566  1.9  2.2  0.873  1.6  3.9   5.5  0.9  0.3   

   BLK  TOV   PF   PTS  Temp TOTAL TEA

Guardamos la temporada ya con su respectivo proceso ETL.

In [27]:
df_filtered.to_csv("Temporada2021_2022.csv",sep=",")

# ANALISIS TEMPORADA 2022 - 2023

In [28]:
df3 = pd.read_csv("nba_salaries.csv") 
print(df3.head(11))

    Unnamed: 0            Player Name    Salary Position  Age     Team  GP  \
0            0          Stephen Curry  48070014       PG   34      GSW  56   
1            1              John Wall  47345760       PG   32      LAC  34   
2            2      Russell Westbrook  47080179       PG   34  LAL/LAC  73   
3            3           LeBron James  44474988       PF   38      LAL  55   
4            4           Kevin Durant  44119845       PF   34  BRK/PHO  47   
5            5           Bradley Beal  43279250       SG   29      WAS  50   
6            6          Kawhi Leonard  42492492       SF   31      LAC  52   
7            7            Paul George  42492492       SF   32      LAC  56   
8            8  Giannis Antetokounmpo  42492492       PF   28      MIL  63   
9            9         Damian Lillard  42492492       PG   32      POR  58   
10          10          Klay Thompson  40600080       SF   32      GSW  69   

    GS    MP    FG   FGA    FG%   3P   3PA    3P%    2P   2PA  

In [29]:
df3.drop(df3.columns[0], axis=1, inplace=True)

In [30]:
print(df3.head(11))

              Player Name    Salary Position  Age     Team  GP  GS    MP  \
0           Stephen Curry  48070014       PG   34      GSW  56  56  34.7   
1               John Wall  47345760       PG   32      LAC  34   3  22.2   
2       Russell Westbrook  47080179       PG   34  LAL/LAC  73  24  29.1   
3            LeBron James  44474988       PF   38      LAL  55  54  35.5   
4            Kevin Durant  44119845       PF   34  BRK/PHO  47  47  35.6   
5            Bradley Beal  43279250       SG   29      WAS  50  50  33.5   
6           Kawhi Leonard  42492492       SF   31      LAC  52  50  33.6   
7             Paul George  42492492       SF   32      LAC  56  56  34.6   
8   Giannis Antetokounmpo  42492492       PF   28      MIL  63  63  32.1   
9          Damian Lillard  42492492       PG   32      POR  58  58  36.3   
10          Klay Thompson  40600080       SF   32      GSW  69  69  33.0   

      FG   FGA    FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FTA  \
0   10.

In [31]:
print(df3.describe())

             Salary         Age          GP          GS          MP  \
count  4.670000e+02  467.000000  467.000000  467.000000  467.000000   
mean   8.416599e+06   25.820128   48.233405   22.650964   19.871306   
std    1.070812e+07    4.275113   24.807740   27.094577    9.548684   
min    5.849000e+03   19.000000    1.000000    0.000000    1.800000   
25%    1.782621e+06   23.000000   31.000000    1.000000   12.500000   
50%    3.722040e+06   25.000000   55.000000    8.000000   19.200000   
75%    1.063354e+07   29.000000   68.500000   45.500000   28.300000   
max    4.807001e+07   42.000000   83.000000   83.000000   41.000000   

               FG         FGA         FG%          3P         3PA         3P%  \
count  467.000000  467.000000  466.000000  467.000000  467.000000  454.000000   
mean     3.351392    7.117773    0.465009    0.996574    2.792719    0.325090   
std      2.457836    5.020700    0.109547    0.880468    2.260794    0.131624   
min      0.000000    0.000000    0.0

Se eliminan las columnas que no sirven. En este caso contenian información no relevante.

In [32]:
df3.drop(df3.columns[30], axis=1, inplace=True)

In [33]:
df3.drop(df3.columns[1], axis=1, inplace=True)

In [34]:
df3.head(11)

Player Name Position  Age     Team  GP  GS    MP    FG   FGA  \
0           Stephen Curry       PG   34      GSW  56  56  34.7  10.0  20.2   
1               John Wall       PG   32      LAC  34   3  22.2   4.1   9.9   
2       Russell Westbrook       PG   34  LAL/LAC  73  24  29.1   5.9  13.6   
3            LeBron James       PF   38      LAL  55  54  35.5  11.1  22.2   
4            Kevin Durant       PF   34  BRK/PHO  47  47  35.6  10.3  18.3   
5            Bradley Beal       SG   29      WAS  50  50  33.5   8.9  17.6   
6           Kawhi Leonard       SF   31      LAC  52  50  33.6   8.6  16.8   
7             Paul George       SF   32      LAC  56  56  34.6   8.2  17.9   
8   Giannis Antetokounmpo       PF   28      MIL  63  63  32.1  11.2  20.3   
9          Damian Lillard       PG   32      POR  58  58  36.3   9.6  20.7   
10          Klay Thompson       SF   32      GSW  69  69  33.0   7.9  18.1   

      FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FTA    FT%  ORB  \
0   0.493  4.9  11.4  0.427   5.1   8.8  0.579  0.614  4.6   5.0  0.915  0.7   
1   0.408  1.0   3.2  0.303   3.1   6.7  0.459  0.457  2.3   3.3  0.681  0.4   
2   0.436  1.2   3.9  0.311   4.7   9.7  0.487  0.481  2.8   4.3  0.656  1.2   
3   0.500  2.2   6.9  0.321   8.9  15.3  0.580  0.549  4.6   5.9  0.768  1.2   
4   0.560  2.0   4.9  0.404   8.3  13.4  0.617  0.614  6.5   7.1  0.919  0.4   
5   0.506  1.6   4.4  0.365   7.3  13.2  0.552  0.551  3.8   4.6  0.842  0.8   
6   0.512  2.0   4.8  0.416   6.6  11.9  0.551  0.572  4.7   5.4  0.871  1.1   
7   0.457  2.8   7.6  0.371   5.4  10.3  0.521  0.536  4.6   5.3  0.871  0.8   
8   0.553  0.7   2.7  0.275  10.5  17.6  0.596  0.572  7.9  12.3  0.645  2.2   
9   0.463  4.2  11.3  0.371   5.4   9.4  0.574  0.564  8.8   9.6  0.914  0.8   
10  0.436  4.4  10.6  0.412   3.6   7.6  0.470  0.556  1.7   1.9  0.879  0.6   

    DRB   TRB  AST  STL  BLK  TOV   PF   PTS  
0   5.4   6.1  6.3  0.9  0.4  3.2  2.1  29.4  
1   2.3   2.7  5.2  0.8  0.4  2.4  1.7  11.4  
2   4.6   5.8  7.5  1.0  0.5  3.5  2.2  15.9  
3   7.1   8.3  6.8  0.9  0.6  3.2  1.6  28.9  
4   6.3   6.7  5.0  0.7  1.4  3.3  2.1  29.1  
5   3.1   3.9  5.4  0.9  0.7  2.9  2.1  23.2  
6   5.4   6.5  3.9  1.4  0.5  1.7  1.6  23.8  
7   5.3   6.1  5.1  1.5  0.4  3.1  2.8  23.8  
8   9.6  11.8  5.7  0.8  0.8  3.9  3.1  31.1  
9   4.0   4.8  7.3  0.9  0.3  3.3  1.9  32.2  
10  3.6   4.1  2.4  0.7  0.4  1.8  1.9  21.9

Verificamos valores nulos

In [35]:
ValoresNull3 = df3.isnull().sum() [df3.isnull().sum()>0]
print(ValoresNull3)

FG%      1
3P%     13
2P%      4
eFG%     1
FT%     23
dtype: int64


In [36]:
# Filtrar filas que contienen al menos un valor nulo
filas_con_nulos = df3[df3.isnull().any(axis=1)]

# Mostrar las primeras filas con valores nulos

np.shape(filas_con_nulos)

(34, 29)

Observamos los valores nulos. Es la primer temporada en presentar valores nulos. A continuación en base a la naturaleza de los nulos fijaremos parámetros para tratarlos.

In [37]:
filas_con_nulos

Player Name Position  Age     Team  GP  GS    MP   FG  FGA    FG%  \
75   Mitchell Robinson        C   24      NYK  59  58  27.0  3.2  4.7  0.671   
133      Mason Plumlee        C   32  LAC/CHO  79  60  26.0  4.2  6.1  0.680   
221      Dylan Windler       SF   26      CLE   3   0   3.3  0.7  1.0  0.667   
233      Mark Williams        C   21      CHO  43  17  19.3  3.7  5.8  0.637   
247     Sterling Brown       SF   27      LAL   4   0   6.0  0.0  1.0  0.000   
254       Joe Wieskamp       SF   23      TOR   9   0   5.6  0.3  1.6  0.214   
264    Bismack Biyombo        C   30      PHO  61  14  14.3  2.0  3.4  0.578   
301     Udoka Azubuike        C   23      UTA  36   4  10.0  1.6  2.0  0.819   
332     Daniel Gafford        C   24      WAS  78  47  20.6  3.7  5.1  0.732   
344   Ryan Arcidiacono       PG   28  NYK/POR  20   4   8.6  0.5  1.9  0.243   
356    Leandro Bolmaro       SF   22      UTA  14   0   4.9  0.2  1.4  0.150   
358       Jared Butler       PG   22      OKC   6   1  12.8  2.5  5.3  0.469   
393  Justin Champagnie       SF   21  BOS/TOR   5   0   6.8  1.0  3.0  0.333   
395        Moses Brown        C   23  BRK/LAC  36   1   8.2  1.7  2.7  0.635   
397          PJ Dozier       SF   26      SAC  16   0   4.9  0.6  2.1  0.303   
403       Trevor Keels       SG   19      NYK   3   0   2.7  0.3  1.3  0.250   
404     Dereon Seabron       SG   22      NOP   5   0   2.4  0.4  1.0  0.400   
408       Malcolm Hill       SF   27      CHI   5   0   1.8  0.4  0.8  0.500   
416        Cole Swider       SF   23      LAL   7   0   5.9  0.4  1.3  0.333   
421      Neemias Queta        C   23      SAC   5   0   5.8  1.2  1.8  0.667   
425       Micah Potter       PF   24      UTA   7   0   7.4  1.4  2.1  0.667   
428       Chima Moneke       SF   27      SAC   2   0   4.0  0.5  1.0  0.500   
436   Alondes Williams       SG   23      BRK   1   0   5.0  0.0  0.0    NaN   
437   Donovan Williams       SG   21      ATL   2   0   2.0  1.0  2.5  0.400   
439        Chris Silva       PF   26      DAL   1   0   3.0  1.0  1.0  1.000   
440       Tyler Dorsey       SF   26      DAL   3   0   2.7  1.3  1.7  0.800   
441        Braxton Key       SF   25      DET   3   0   3.0  0.3  0.3  1.000   
444       Devon Dotson       PG   23      WAS   6   0   8.8  0.2  1.7  0.100   
449        Xavier Moon       SG   28      LAC   4   0   5.0  0.8  2.3  0.333   
453      Frank Jackson       PG   24      UTA   1   0   5.0  0.0  3.0  0.000   
454      Deonte Burton       PF   29      SAC   2   0   3.0  0.0  1.0  0.000   
458     Jordan Schakel       SF   24      WAS   2   0   3.0  0.5  1.0  0.500   
459      Stanley Umude       SG   23      DET   1   0   2.0  0.0  1.0  0.000   
466      Jacob Gilyard       PG   24      MEM   1   0  41.0  1.0  3.0  0.333   

      3P  3PA    3P%   2P  2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  \
75   0.0  0.0    NaN  3.2  4.7  0.671  0.671  1.0  2.1  0.484  4.5  4.9  9.4   
133  0.0  0.0    NaN  4.2  6.1  0.680  0.680  2.5  4.0  0.636  2.9  6.0  8.9   
221  0.3  0.7  0.500  0.3  0.3  1.000  0.833  0.0  0.0    NaN  0.0  0.0  0.0   
233  0.0  0.0    NaN  3.7  5.8  0.637  0.637  1.6  2.3  0.691  2.2  4.9  7.1   
247  0.0  0.5  0.000  0.0  0.5  0.000  0.000  0.0  0.0    NaN  0.8  1.3  2.0   
254  0.3  1.3  0.250  0.0  0.2  0.000  0.321  0.0  0.0    NaN  0.0  0.4  0.4   
264  0.0  0.0    NaN  2.0  3.4  0.578  0.578  0.4  1.1  0.357  1.5  2.8  4.3   
301  0.0  0.0    NaN  1.6  2.0  0.819  0.819  0.2  0.6  0.350  0.9  2.4  3.3   
332  0.0  0.0    NaN  3.7  5.1  0.732  0.732  1.6  2.4  0.679  2.1  3.5  5.6   
344  0.4  1.2  0.348  0.1  0.7  0.071  0.351  0.0  0.0    NaN  0.0  0.8  0.8   
356  0.0  0.3  0.000  0.2  1.1  0.188  0.150  0.0  0.0    NaN  0.3  0.2  0.5   
358  1.2  2.3  0.500  1.3  3.0  0.444  0.578  0.0  0.0    NaN  0.2  0.5  0.7   
393  0.2  1.0  0.200  0.8  2.0  0.400  0.367  0.0  0.0    NaN  0.6  1.0  1.6   
395  0.0  0.0    NaN  1.7  2.7  0.635  0.635  0.9  2.0  0.458  1.6  2.3  3.9   
397  0.1  

Consultaremos los datos oficiales de la página de la nba (www.nba.com)

Esto lo haremos columna por columna debido a la delicada naturaleza de cada estadistica.

In [38]:
columnas_a_imprimir = ['Player Name','FG%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    FG%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  0.667
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.214
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.243
356    Leandro Bolmaro  0.150
358       Jared Butler  0.469
393  Justin Champagnie  0.333
395        Moses Brown  0.635
397          PJ Dozier  0.303
403       Trevor Keels  0.250
404     Dereon Seabron  0.400
408       Malcolm Hill  0.500
416        Cole Swider  0.333
421      Neemias Queta  0.667
425       Micah Potter  0.667
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.400
439        Chris Silva  1.000
440       Tyler Dorsey  0.800
441        Braxton Key  1.000
444       Devon Dotson  0.100
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.500
459      S

 Alondes Williams el unico valor faltante de aquí no tiene su estadística disponible en la página de la NBA entonces para no generar data errónea lo mejor será colocarle 0.

In [39]:
# Rellenar valores faltantes de Alondes Williams con 0 en la fila con índice 436
alondes = [436]
df3.loc[alondes, 'FG%'] = df3.loc[alondes, 'FG%'].fillna(0)

In [40]:
columnas_a_imprimir = ['Player Name','3P%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    3P%
75   Mitchell Robinson    NaN
133      Mason Plumlee    NaN
221      Dylan Windler  0.500
233      Mark Williams    NaN
247     Sterling Brown  0.000
254       Joe Wieskamp  0.250
264    Bismack Biyombo    NaN
301     Udoka Azubuike    NaN
332     Daniel Gafford    NaN
344   Ryan Arcidiacono  0.348
356    Leandro Bolmaro  0.000
358       Jared Butler  0.500
393  Justin Champagnie  0.200
395        Moses Brown    NaN
397          PJ Dozier  0.125
403       Trevor Keels  0.250
404     Dereon Seabron    NaN
408       Malcolm Hill  0.333
416        Cole Swider  0.375
421      Neemias Queta    NaN
425       Micah Potter  0.571
428       Chima Moneke    NaN
436   Alondes Williams    NaN
437   Donovan Williams  0.000
439        Chris Silva    NaN
440       Tyler Dorsey  0.500
441        Braxton Key    NaN
444       Devon Dotson  0.250
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  1.000
459      S

En el caso de estos valores faltantes, si figuran en la página oficial de NBA pero aparecen como 0%.

Esto refleja que hay jugadores que no tuvieron oportunidades de hacer tiros de 3 puntos.

 Procedo a rellenarlos con 0.

In [41]:
indices = [75,133,233,264,301,332,395,404,421,428,436,441,439]
df3.loc[indices, '3P%'] = df3.loc[indices, '3P%'].fillna(0)

In [42]:
columnas_a_imprimir = ['Player Name','2P%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    2P%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  1.000
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.000
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.071
356    Leandro Bolmaro  0.188
358       Jared Butler  0.444
393  Justin Champagnie  0.400
395        Moses Brown  0.635
397          PJ Dozier  0.471
403       Trevor Keels    NaN
404     Dereon Seabron  0.400
408       Malcolm Hill  1.000
416        Cole Swider  0.000
421      Neemias Queta  0.667
425       Micah Potter  0.750
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.667
439        Chris Silva  1.000
440       Tyler Dorsey  1.000
441        Braxton Key  1.000
444       Devon Dotson  0.000
449        Xavier Moon  0.333
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.000
459      S

Stanley Umude: No tuvo estadísticas de la NBA para la temporada 2022-2023, ya que no jugó en partidos de temporada regular.

Jacob Gilyard: Similar a Umude, Jacob Gilyard no jugó en la NBA durante la temporada 2022-2023, por lo que no tiene estadísticas de 2P%.

Trevor Keels: Trevor Keels, quien también fue novato en la temporada 2022-2023, jugó para los New York Knicks y tuvo un porcentaje de tiros de campo de dos puntos (2P%) de 33.3% en los partidos que disputó.

Alondes Williams: Alondes Williams tampoco tuvo estadísticas en la NBA durante la temporada 2022-2023, ya que no participó en partidos de temporada regular.

In [43]:
indices = [466,459,436]
df3.loc[indices, '2P%'] = df3.loc[indices, '2P%'].fillna(0)
trevor_keels = [403]
df3.loc[trevor_keels, '2P%'] = df3.loc[trevor_keels, '2P%'].fillna(0.333)

In [44]:
columnas_a_imprimir = ['Player Name','eFG%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name   eFG%
75   Mitchell Robinson  0.671
133      Mason Plumlee  0.680
221      Dylan Windler  0.833
233      Mark Williams  0.637
247     Sterling Brown  0.000
254       Joe Wieskamp  0.321
264    Bismack Biyombo  0.578
301     Udoka Azubuike  0.819
332     Daniel Gafford  0.732
344   Ryan Arcidiacono  0.351
356    Leandro Bolmaro  0.150
358       Jared Butler  0.578
393  Justin Champagnie  0.367
395        Moses Brown  0.635
397          PJ Dozier  0.333
403       Trevor Keels  0.375
404     Dereon Seabron  0.400
408       Malcolm Hill  0.625
416        Cole Swider  0.500
421      Neemias Queta  0.667
425       Micah Potter  0.800
428       Chima Moneke  0.500
436   Alondes Williams    NaN
437   Donovan Williams  0.400
439        Chris Silva  1.000
440       Tyler Dorsey  0.900
441        Braxton Key  1.000
444       Devon Dotson  0.150
449        Xavier Moon  0.389
453      Frank Jackson  0.000
454      Deonte Burton  0.000
458     Jordan Schakel  0.750
459      S

In [45]:
alondes = [436]
df3.loc[alondes, 'eFG%'] = df3.loc[alondes, 'eFG%'].fillna(0)

In [46]:
columnas_a_imprimir = ['Player Name','FT%']
print(filas_con_nulos[columnas_a_imprimir])

           Player Name    FT%
75   Mitchell Robinson  0.484
133      Mason Plumlee  0.636
221      Dylan Windler    NaN
233      Mark Williams  0.691
247     Sterling Brown    NaN
254       Joe Wieskamp    NaN
264    Bismack Biyombo  0.357
301     Udoka Azubuike  0.350
332     Daniel Gafford  0.679
344   Ryan Arcidiacono    NaN
356    Leandro Bolmaro    NaN
358       Jared Butler    NaN
393  Justin Champagnie    NaN
395        Moses Brown  0.458
397          PJ Dozier    NaN
403       Trevor Keels    NaN
404     Dereon Seabron    NaN
408       Malcolm Hill    NaN
416        Cole Swider    NaN
421      Neemias Queta  0.000
425       Micah Potter    NaN
428       Chima Moneke  0.000
436   Alondes Williams    NaN
437   Donovan Williams    NaN
439        Chris Silva    NaN
440       Tyler Dorsey    NaN
441        Braxton Key  1.000
444       Devon Dotson    NaN
449        Xavier Moon    NaN
453      Frank Jackson    NaN
454      Deonte Burton    NaN
458     Jordan Schakel    NaN
459      S

Estas fueron las stats oficiales (FT%) de la pagina NBA para cada jugador.

Dylan Windler: 70.4%

Sterling Brown: 78.6%

Joe Wieskamp: 71.4%

Ryan Arcidiacono: 80.0%

Leandro Bolmaro: 68.8%

Jared Butler: 76.2%

Justin Champagnie: 76.5%

P.J. Dozier: 69.4%

Trevor Keels: 63.6%

Dereon Seabron: 77.8%

Malcolm Hill: 81.8%

Cole Swider: — (No jugó en la NBA durante la temporada 2022-2023)

Micah Potter: — (No jugó en la NBA durante la temporada 2022-2023)

Alondes Williams: — (No jugó en la NBA durante la temporada 2022-2023)

Donovan Williams: — (No jugó en la NBA durante la temporada 2022-2023)

Chris Silva: 78.9%

Tyler Dorsey: 83.3%

Devon Dotson: 80.0%

Xavier Moon: 82.8%

Frank Jackson: 83.0%

Deonte Burton: 80.0%

Jordan Schakel: — (No jugó en la NBA durante la temporada 2022-2023)

Jacob Gilyard: — (No jugó en la NBA durante la temporada 2022-2023)


In [47]:

# Diccionario con valores de reemplazo
reemplazos = {
    'Dylan Windler': 0.704,
    'Sterling Brown': 0.786,
    'Joe Wieskamp': 0.714,
    'Ryan Arcidiacono': 0.800,
    'Leandro Bolmaro': 0.688,
    'Jared Butler': 0.762,
    'Justin Champagnie': 0.765,
    'PJ Dozier': 0.694,
    'Trevor Keels': 0.636,
    'Dereon Seabron': 0.778,
    'Malcolm Hill': 0.818,
    'Cole Swider': 0,  # No jugó
    'Micah Potter': 0,  # No jugó
    'Alondes Williams': 0,  # No jugó
    'Donovan Williams': 0,  # No jugó
    'Chris Silva': 0.789,
    'Tyler Dorsey': 0.833,
    'Devon Dotson': 0.800,
    'Xavier Moon': 0.828,
    'Frank Jackson': 0.830,
    'Deonte Burton': 0.800,
    'Jordan Schakel': 0,  # No jugó
    'Jacob Gilyard': 0  # No jugó
}

# Rellenar los valores faltantes en la columna 'FT%' con el diccionario de reemplazo
df3['FT%'] = df3.apply(lambda row: reemplazos.get(row['Player Name'], row['FT%']), axis=1)

Checkeo nuevamente valores faltantes:

In [48]:
ValoresNull3 = df3.isnull().sum() [df3.isnull().sum()>0]
print(ValoresNull3)

Series([], dtype: int64)


In [49]:
# Lista de columnas para las que se reemplazarán los valores nulos con la media
columnas_objetivo = ['FG%', '3P%', '2P%', 'eFG%', 'FT%']

# Reemplazar los valores nulos de las columnas especificadas por su media
for columna in columnas_objetivo:
    media_columna = df3[columna].mean()
    df3[columna] = df3[columna].fillna(media_columna)

# Mostrar las primeras filas del DataFrame para verificar
print(df3.head(11))

              Player Name Position  Age     Team  GP  GS    MP    FG   FGA  \
0           Stephen Curry       PG   34      GSW  56  56  34.7  10.0  20.2   
1               John Wall       PG   32      LAC  34   3  22.2   4.1   9.9   
2       Russell Westbrook       PG   34  LAL/LAC  73  24  29.1   5.9  13.6   
3            LeBron James       PF   38      LAL  55  54  35.5  11.1  22.2   
4            Kevin Durant       PF   34  BRK/PHO  47  47  35.6  10.3  18.3   
5            Bradley Beal       SG   29      WAS  50  50  33.5   8.9  17.6   
6           Kawhi Leonard       SF   31      LAC  52  50  33.6   8.6  16.8   
7             Paul George       SF   32      LAC  56  56  34.6   8.2  17.9   
8   Giannis Antetokounmpo       PF   28      MIL  63  63  32.1  11.2  20.3   
9          Damian Lillard       PG   32      POR  58  58  36.3   9.6  20.7   
10          Klay Thompson       SF   32      GSW  69  69  33.0   7.9  18.1   

      FG%   3P   3PA    3P%    2P   2PA    2P%   eFG%   FT   FT

In [50]:
# Mostrar el valor de la columna 'FT%' en la fila 221
valor_fila_221 = df3.loc[221, 'FT%']
print(f"Valor en la fila 221 de 'FT%': {valor_fila_221}")

# Mostrar el valor de la columna 'FT%' en la fila 247
valor_fila_247 = df3.loc[247, 'FT%']
print(f"Valor en la fila 247 de 'FT%': {valor_fila_247}")

Valor en la fila 221 de 'FT%': 0.704
Valor en la fila 247 de 'FT%': 0.786


In [51]:
nuevos_nombres = {
    'Player Name': 'Player',
    'Position': 'Pos'
}

# Renombrar las columnas especificadas
df3.rename(columns=nuevos_nombres, inplace=True)


# Mostrar las primeras filas del DataFrame para verificar los cambios
print(df3.head())

              Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%   3P  \
0      Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1          John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2  Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3       LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4       Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560  2.0   

    3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
0  11.4  0.427  5.1   8.8  0.579  0.614  4.6  5.0  0.915  0.7  5.4  6.1  6.3   
1   3.2  0.303  3.1   6.7  0.459  0.457  2.3  3.3  0.681  0.4  2.3  2.7  5.2   
2   3.9  0.311  4.7   9.7  0.487  0.481  2.8  4.3  0.656  1.2  4.6  5.8  7.5   
3   6.9  0.321  8.9  15.3  0.580  0.549  4.6  5.9  0.768  1.2  7.1  8.3  6.8   
4   4.9  0.404  8.3  13.4  0.617  0.614  6.5  7.1  0.919  0.4  6.3  6.7  5.0   

   STL  BLK  TOV   PF   PTS  
0  0.9  

In [52]:
# Año que deseas agregar
Temporada3 = 2023

# Agregar una nueva columna 'Te,`prada' con el valor del año especificado
df3['Temp'] = Temporada3

# Mostrar las primeras filas del DataFrame para verificar
print(df3.head())

              Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%   3P  \
0      Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1          John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2  Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3       LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4       Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560  2.0   

    3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
0  11.4  0.427  5.1   8.8  0.579  0.614  4.6  5.0  0.915  0.7  5.4  6.1  6.3   
1   3.2  0.303  3.1   6.7  0.459  0.457  2.3  3.3  0.681  0.4  2.3  2.7  5.2   
2   3.9  0.311  4.7   9.7  0.487  0.481  2.8  4.3  0.656  1.2  4.6  5.8  7.5   
3   6.9  0.321  8.9  15.3  0.580  0.549  4.6  5.9  0.768  1.2  7.1  8.3  6.8   
4   4.9  0.404  8.3  13.4  0.617  0.614  6.5  7.1  0.919  0.4  6.3  6.7  5.0   

   STL  BLK  TOV   PF   PTS  Temp  
0 

In [53]:
# Duplicar la columna 'Tm' y renombrarla a 'Total Team'
df3['TOTAL TEAM'] = df3['Team']

# Mostrar las primeras filas del DataFrame para verificar
print(df3.head())

              Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%   3P  \
0      Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493  4.9   
1          John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408  1.0   
2  Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436  1.2   
3       LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500  2.2   
4       Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560  2.0   

    3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
0  11.4  0.427  5.1   8.8  0.579  0.614  4.6  5.0  0.915  0.7  5.4  6.1  6.3   
1   3.2  0.303  3.1   6.7  0.459  0.457  2.3  3.3  0.681  0.4  2.3  2.7  5.2   
2   3.9  0.311  4.7   9.7  0.487  0.481  2.8  4.3  0.656  1.2  4.6  5.8  7.5   
3   6.9  0.321  8.9  15.3  0.580  0.549  4.6  5.9  0.768  1.2  7.1  8.3  6.8   
4   4.9  0.404  8.3  13.4  0.617  0.614  6.5  7.1  0.919  0.4  6.3  6.7  5.0   

   STL  BLK  TOV   PF   PTS  Temp TOTA

Guardamos el proceso etl de la temporada 2022-2023.

In [54]:
df3.to_csv("Temporada2022_2023.csv",sep=",")

### ANALISIS SALARIOS 22-23

In [55]:
df4 = pd.read_csv("nba_2022-23_all_stats_with_salary.csv") 
print(df4.head(11))

    Unnamed: 0            Player Name    Salary Position  Age     Team  GP  \
0            0          Stephen Curry  48070014       PG   34      GSW  56   
1            1              John Wall  47345760       PG   32      LAC  34   
2            2      Russell Westbrook  47080179       PG   34  LAL/LAC  73   
3            3           LeBron James  44474988       PF   38      LAL  55   
4            4           Kevin Durant  44119845       PF   34  BRK/PHO  47   
5            5           Bradley Beal  43279250       SG   29      WAS  50   
6            6          Kawhi Leonard  42492492       SF   31      LAC  52   
7            7            Paul George  42492492       SF   32      LAC  56   
8            8  Giannis Antetokounmpo  42492492       PF   28      MIL  63   
9            9         Damian Lillard  42492492       PG   32      POR  58   
10          10          Klay Thompson  40600080       SF   32      GSW  69   

    GS    MP    FG   FGA    FG%   3P   3PA    3P%    2P   2PA  

In [56]:
df4.drop(df4.columns[3:51], axis=1, inplace=True)



In [57]:
df4.drop(df4.columns[0], axis=1, inplace=True)

In [58]:
df4.drop(df4.columns[2], axis=1, inplace=True)

Lo que hicimos fue quedarnos con las columnas de Salario y nombre de jugador unicamente.

In [59]:
print(df4.head(11))

              Player Name    Salary
0           Stephen Curry  48070014
1               John Wall  47345760
2       Russell Westbrook  47080179
3            LeBron James  44474988
4            Kevin Durant  44119845
5            Bradley Beal  43279250
6           Kawhi Leonard  42492492
7             Paul George  42492492
8   Giannis Antetokounmpo  42492492
9          Damian Lillard  42492492
10          Klay Thompson  40600080


In [60]:
ValoresNull4 = df4.isnull().sum() [df4.isnull().sum()>0]
print(ValoresNull4)

Series([], dtype: int64)


Se revisaron valores nulos.

In [61]:
df4.isnull().sum()

Player Name    0
Salary         0
dtype: int64

In [62]:
df4.rename(columns={'Player Name': 'Player'}, inplace=True)

In [63]:
# Año que deseas agregar
Temporada4 = 2023

# Agregar una nueva columna 'Año' con el valor del año especificado
df4['Temp'] = Temporada4

# Mostrar las primeras filas del DataFrame para verificar
print(df4.head())

              Player    Salary  Temp
0      Stephen Curry  48070014  2023
1          John Wall  47345760  2023
2  Russell Westbrook  47080179  2023
3       LeBron James  44474988  2023
4       Kevin Durant  44119845  2023


Se guarda el dataset de salarios ya terminado.

In [64]:
df4.to_csv("Salarios.csv",sep=",")

# Se comienza a preparar todo para realizar el merge de temporadas.

In [65]:
df2021_2022 = pd.read_csv("Temporada2021_2022.csv") 
print(df3.head(11))

                   Player Pos  Age     Team  GP  GS    MP    FG   FGA    FG%  \
0           Stephen Curry  PG   34      GSW  56  56  34.7  10.0  20.2  0.493   
1               John Wall  PG   32      LAC  34   3  22.2   4.1   9.9  0.408   
2       Russell Westbrook  PG   34  LAL/LAC  73  24  29.1   5.9  13.6  0.436   
3            LeBron James  PF   38      LAL  55  54  35.5  11.1  22.2  0.500   
4            Kevin Durant  PF   34  BRK/PHO  47  47  35.6  10.3  18.3  0.560   
5            Bradley Beal  SG   29      WAS  50  50  33.5   8.9  17.6  0.506   
6           Kawhi Leonard  SF   31      LAC  52  50  33.6   8.6  16.8  0.512   
7             Paul George  SF   32      LAC  56  56  34.6   8.2  17.9  0.457   
8   Giannis Antetokounmpo  PF   28      MIL  63  63  32.1  11.2  20.3  0.553   
9          Damian Lillard  PG   32      POR  58  58  36.3   9.6  20.7  0.463   
10          Klay Thompson  SF   32      GSW  69  69  33.0   7.9  18.1  0.436   

     3P   3PA    3P%    2P   2PA    2P%

In [66]:
# Cargar los tres archivos CSV en DataFrames
df1final = pd.read_csv('Temporada2020_2021.csv')
df2final = pd.read_csv('Temporada2021_2022.csv')
df3final = pd.read_csv('Temporada2022_2023.csv')

# Concatenar los DataFrames
df_combined = pd.concat([df1final, df2final, df3final], ignore_index=True)
df_combined = df_combined.drop('Unnamed: 0', axis=1)
# Guardar el DataFrame combinado en un nuevo archivo CSV
df_combined.to_csv('TempFinal.csv', index=False)

Se realizó el merge y se procede a guardarlo en un .csv

In [67]:
df_final = pd.read_csv('TempFinal.csv')


In [68]:
df_final.head(11)

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%  \
0           Precious Achiuwa  PF   21  MIA  28   2  14.6  2.6   4.4  0.590   
1               Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125   
2               Steven Adams   C   27  NOP  27  27  28.1  3.5   5.8  0.603   
3                Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573   
4          LaMarcus Aldridge   C   35  SAS  18  18  26.7  5.9  12.5  0.476   
5          Ty-Shon Alexander  SG   22  PHO   3   0   2.7  0.0   1.0  0.000   
6   Nickeil Alexander-Walker  SG   22  NOP  23   3  19.2  3.3   8.2  0.410   
7              Grayson Allen  SG   25  MEM  19   8  23.9  3.2   7.4  0.429   
8              Jarrett Allen   C   22  TOT  28  10  26.2  4.4   6.8  0.642   
9              Jarrett Allen   C   22  BRK  12   5  26.7  3.7   5.4  0.677   
10             Jarrett Allen   C   22  CLE  16   5  25.9  4.9   7.8  0.624   

     3P  3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB   TRB  \
0   0.0  0.0  0.000  2.6   4.4  0.590  0.590  1.3  2.4  0.561  1.3  2.7   4.0   
1   0.0  0.3  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5   0.5   
2   0.0  0.0  0.000  3.5   5.7  0.606  0.603  1.1  2.3  0.468  4.3  4.6   8.9   
3   0.1  0.2  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3   9.2   
4   1.3  3.7  0.358  4.6   8.8  0.525  0.529  0.9  1.2  0.762  0.8  3.5   4.3   
5   0.0  0.3  0.000  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3   0.3   
6   1.0  3.8  0.276  2.3   4.4  0.525  0.473  1.1  1.4  0.781  0.2  2.4   2.7   
7   2.3  5.3  0.436  0.8   2.1  0.410  0.586  1.7  1.9  0.892  0.4  2.5   2.9   
8   0.0  0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1   9.0   
9   0.0  0.0  0.000  3.7   5.4  0.677  0.677  3.8  5.1  0.754  3.2  7.3  10.4   
10  0.1  0.3  0.250  4.8   7.6  0.636  0.628  3.4  4.4  0.761  2.8  5.2   7.9   

    AST  STL  BLK  TOV   PF   PTS  Temp TOTAL TEAM  
0   0.6  0.4  0.5  1.0  1.9   6.5  2021        MIA  
1   0.3  0.0  0.0  0.0  0.2   0.3  2021        MIL  
2   2.1  1.0  0.6  1.7  1.9   8.0  2021        NOP  
3   5.3  1.0  1.0  3.0  2.6  19.9  2021        MIA  
4   1.9  0.4  0.9  0.9  1.5  14.1  2021        SAS  
5   0.3  0.0  0.0  0.0  0.3   0.0  2021        PHO  
6   2.0  1.1  0.3  1.3  1.7   8.8  2021        NOP  
7   2.1  1.0  0.2  1.1  1.3  10.4  2021        MEM  
8   1.6  0.5  1.6  1.5  1.6  12.3  2021        TOT  
9   1.7  0.6  1.6  1.8  1.8  11.2  2021        BRK  
10  1.6  0.4  1.7  1.3  1.4  13.2  2021        CLE

Ahora arreglamos los datos duplicados, unificando los equipos totales en los que jugaron en la temporada en la columna TOTAL TEAM.

In [69]:
# Crear la columna TOTAL TEAMS concatenando los nombres de los equipos
df_final['TOTAL TEAM'] = df_final.groupby(['Player'])['Team'].transform(lambda x: '/'.join(x[x != 'TOT']))

# Filtrar solo las filas donde el equipo es 'TOT'
df_tot = df_final[df_final['Team'] == 'TOT'].copy()

pd.options.display.max_columns = 60
pd.options.display.max_columns

# Ver el resultado
df_tot.head(20)


Player    Pos  Age Team  GP  GS    MP   FG   FGA  \
8               Jarrett Allen      C   22  TOT  28  10  26.2  4.4   6.8   
181              James Harden     SG   31  TOT  24  24  37.6  7.6  15.8   
255            Rodions Kurucs  SF-PF   22  TOT  14   0   5.2  0.4   1.4   
265                  Alex Len      C   27  TOT  20   6  14.6  2.4   3.8   
354            Victor Oladipo     SG   28  TOT  20  20  32.0  7.0  17.5   
382            Taurean Prince     PF   26  TOT  27  10  21.9  2.8   7.2   
403              Derrick Rose     PG   32  TOT  20   0  22.3  5.0  11.7   
425          Dennis Smith Jr.     PG   23  TOT   7   0  12.1  1.6   4.7   
502  Nickeil Alexander-Walker     SG   23  TOT  65  21  22.6  3.9  10.5   
506           Justin Anderson     SF   28  TOT  16   6  19.8  2.3   5.9   
515             D.J. Augustin     PG   34  TOT  55   2  16.1  1.7   4.0   
520         Marvin Bagley III     PF   22  TOT  48  25  23.9  4.7   9.3   
540           DeAndre' Bembry     SF   27  TOT  56  20  18.3  2.2   3.9   
541             D?vis Bert?ns     PF   29  TOT  56   0  14.4  1.8   4.9   
566             Armoni Brooks  SG-PG   23  TOT  54  11  15.6  1.8   5.4   
569         Charlie Brown Jr.     SG   24  TOT  22   2   8.0  0.5   1.8   
570        Chaundee Brown Jr.  SG-SF   23  TOT   5   2  20.8  2.0   6.4   
573               Moses Brown      C   22  TOT  40   6   8.6  1.6   2.7   
590          Vernon Carey Jr.      C   20  TOT   7   1   6.3  1.0   1.9   
591              Jevon Carter     PG   26  TOT  66   3  13.7  1.5   3.8   

       FG%   3P  3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  \
8    0.642  0.0  0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9   
181  0.482  3.0  7.9  0.384  4.6   7.9  0.579  0.578  6.2  6.9  0.897  0.8   
255  0.300  0.2  1.0  0.214  0.2   0.4  0.500  0.375  0.1  0.1  0.500  0.1   
265  0.627  0.3  1.0  0.316  2.1   2.8  0.732  0.667  1.0  1.7  0.588  0.8   
354  0.401  2.4  7.3  0.329  4.6  10.2  0.453  0.470  2.7  3.7  0.740  0.2   
382  0.392  1.4  3.5  0.400  1.4   3.7  0.384  0.490  1.7  2.0  0.849  0.4   
403  0.427  0.8  2.4  0.333  4.2   9.3  0.452  0.462  2.7  3.2  0.841  0.4   
425  0.333  0.3  1.7  0.167  1.3   3.0  0.429  0.364  1.1  1.4  0.800  0.3   
502  0.372  1.6  5.2  0.311  2.3   5.3  0.433  0.449  1.2  1.7  0.743  0.6   
506  0.379  0.9  3.7  0.254  1.3   2.3  0.583  0.458  0.9  1.2  0.789  0.3   
515  0.423  1.2  3.0  0.415  0.5   1.1  0.448  0.577  0.7  0.8  0.884  0.2   
520  0.504  0.5  2.0  0.237  4.2   7.3  0.579  0.530  1.4  2.1  0.663  2.2   
540  0.561  0.3  0.6  0.417  1.9   3.3  0.589  0.595  0.4  0.6  0.600  0.8   
541  0.360  1.4  4.1  0.335  0.4   0.8  0.489  0.500  0.6  0.7  0.900  0.2   
566  0.339  1.4  4.6  0.297  0.4   0.7  0.600  0.467  0.3  0.4  0.857  0.4   
569  0.256  0.0  0.5  0.083  0.4   1.2  0.333  0.269  0.4  0.5  0.900  0.6   
570  0.313  1.2  3.6  0.333  0.8   2.8  0.286  0.406  1.0  1.2  0.833  0.2   
573  0.593  0.0  0.0  0.000  1.6   2.7  0.593  0.593  1.1  1.8  0.597  1.0   
590  0.538  0.0  0.0  0.000  1.0   1.9  0.538  0.538  0.9  1.9  0.462  1.1   
591  0.387  1.0  2.6  0.388  0.5   1.3  0.386  0.518  0.2  0.3  0.833  0.2   

     DRB  TRB   AST  STL  BLK  TOV   PF   PTS  Temp       TOTAL TEAM  
8    6.1  9.0   1.6  0.5  1.6  1.5  1.6  12.3  2021  BRK/CLE/CLE/CLE  
181  6.5  7.3  11.3  1.0  0.8  4.3  2.3  24.5  2021      HOU/BRK/PHI  
255  0.7  0.9   0.3  0.4  0.1  0.5  0.6   1.1  2021          BRK/HOU  
265  2.8  3.5   1.1  0.4  1.1  1.2  1.9   6.0  2021  TOR/WAS/SAC/SAC  
354  5.0  5.2   4.6  1.4  0.4  2.2  2.4  19.1  2021  IND/HOU/MIA/MIA  
382  3.2  3.6   1.7  0.7  0.7  1.1  1.9   8.7  2021  BRK/CLE/MIN/MIN  
403  1.8  2.2   4.0  1.2  0.4  1.9  1.1  13.5  2021  DET/NYK/NYK/NYK  
425  0.9  1.1   1.6  0.4  0.1  0.7  0.7   4.6  2021      NYK/DET/POR  
502  2.3  2.9   2.4  0.7  0.4  1.4  1.6  10.6  2022      NOP/MIN/UTA  
506  2.6  2.9   2.1  0.5  0.4  0.5  1.4   6.4  2022                   
515  1.0  1.2   1.9  0.3  0

In [70]:
# Identificar los jugadores que han jugado en más de un equipo en una temporada
multi_team_players = df_final.groupby(['Player'])['Team'].transform('nunique') > 1

# Filtrar para quedarte solo con la fila "TOT" para los jugadores con múltiples equipos
df_filtered = df_final[~multi_team_players | (df_final['Team'] == 'TOT')]
np.shape(df_filtered)


(870, 31)

In [71]:
df_filtered.head()

Player Pos  Age Team  GP  GS    MP   FG   FGA    FG%   3P  3PA  \
1        Jaylen Adams  PG   24  MIL   6   0   2.8  0.2   1.3  0.125  0.0  0.3   
3         Bam Adebayo   C   23  MIA  26  26  33.6  7.4  12.9  0.573  0.1  0.2   
5   Ty-Shon Alexander  SG   22  PHO   3   0   2.7  0.0   1.0  0.000  0.0  0.3   
8       Jarrett Allen   C   22  TOT  28  10  26.2  4.4   6.8  0.642  0.0  0.1   
11    Al-Farouq Aminu  PF   30  ORL   2   0   8.0  0.5   0.5  1.000  0.5  0.5   

     3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  STL  \
1   0.00  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5  0.3  0.0   
3   0.40  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2  5.3  1.0   
5   0.00  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3  0.3  0.3  0.0   
8   0.25  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1  9.0  1.6  0.5   
11  1.00  0.0   0.0  0.000  1.500  0.0  1.0  0.000  0.5  1.5  2.0  1.0  0.0   

    BLK  TOV   PF   PTS  Temp       TOTAL TEAM  
1   0.0  0.0  0.2   0.3  2021              MIL  
3   1.0  3.0  2.6  19.9  2021      MIA/MIA/MIA  
5   0.0  0.0  0.3   0.0  2021              PHO  
8   1.6  1.5  1.6  12.3  2021  BRK/CLE/CLE/CLE  
11  0.0  1.0  0.0   1.5  2021              ORL

Se le agregan corchetes al principio y al final, dejando los valores de los equipos en el medio, a las columnas Team y TOTAL TEAM.

Esto es para no tener problemas de reconocimiento a la hora de guardar el CSV y enviarlo a SQL, debido a que los datos se separan con '/' en esas columnas.

In [72]:
df_filtered['Team'] = df_filtered['Team'].apply(lambda x: f"[{x}]")
df_filtered['TOTAL TEAM'] = df_filtered['TOTAL TEAM'].apply(lambda x: f"[{x}]")

C:\Users\Agus\AppData\Local\Temp\ipykernel_12692\3079955550.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['Team'] = df_filtered['Team'].apply(lambda x: f"[{x}]")
C:\Users\Agus\AppData\Local\Temp\ipykernel_12692\3079955550.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['TOTAL TEAM'] = df_filtered['TOTAL TEAM'].apply(lambda x: f"[{x}]")


In [73]:
df_filtered

Player Pos  Age   Team  GP  GS    MP   FG   FGA    FG%   3P  \
1          Jaylen Adams  PG   24  [MIL]   6   0   2.8  0.2   1.3  0.125  0.0   
3           Bam Adebayo   C   23  [MIA]  26  26  33.6  7.4  12.9  0.573  0.1   
5     Ty-Shon Alexander  SG   22  [PHO]   3   0   2.7  0.0   1.0  0.000  0.0   
8         Jarrett Allen   C   22  [TOT]  28  10  26.2  4.4   6.8  0.642  0.0   
11      Al-Farouq Aminu  PF   30  [ORL]   2   0   8.0  0.5   0.5  1.000  0.5   
...                 ...  ..  ...    ...  ..  ..   ...  ...   ...    ...  ...   
1564      Justin Minaya  SF   23  [POR]   4   0  22.3  1.8   5.8  0.304  0.8   
1565       Kobi Simmons  SG   25  [CHO]   5   0   5.6  0.2   1.2  0.167  0.2   
1566          Gabe York  SG   29  [IND]   3   0  18.7  2.7   7.0  0.381  2.0   
1567       RaiQuan Gray  PF   23  [BRK]   1   0  35.0  6.0  12.0  0.500  2.0   
1568      Jacob Gilyard  PG   24  [MEM]   1   0  41.0  1.0   3.0  0.333  1.0   

      3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  \
1     0.3  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5   
3     0.2  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2   
5     0.3  0.000  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3  0.3   
8     0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1  9.0   
11    0.5  1.000  0.0   0.0  0.000  1.500  0.0  1.0  0.000  0.5  1.5  2.0   
...   ...    ...  ...   ...    ...    ...  ...  ...    ...  ...  ...  ...   
1564  3.0  0.250  1.0   2.8  0.364  0.370  0.0  0.3  0.000  0.8  3.0  3.8   
1565  1.0  0.200  0.0   0.2  0.000  0.250  0.4  0.4  1.000  0.2  0.6  0.8   
1566  6.0  0.333  0.7   1.0  0.667  0.524  0.7  0.7  1.000  0.0  2.0  2.0   
1567  5.0  0.400  4.0   7.0  0.571  0.583  2.0  2.0  1.000  3.0  6.0  9.0   
1568  3.0  0.333  0.0   0.0  0.000  0.500  0.0  0.0  0.000  0.0  4.0  4.0   

      AST  STL  BLK  TOV   PF   PTS  Temp         TOTAL TEAM  
1     0.3  0.0  0.0  0.0  0.2   0.3  2021              [MIL]  
3     5.3  1.0  1.0  3.0  2.6  19.9  2021      [MIA/MIA/MIA]  
5     0.3  0.0  0.0  0.0  0.3   0.0  2021              [PHO]  
8     1.6  0.5  1.6  1.5  1.6  12.3  2021  [BRK/CLE/CLE/CLE]  
11    1.0  0.0  0.0  1.0  0.0   1.5  2021              [ORL]  
...   ...  ...  ...  ...  ...   ...   ...                ...  
1564  1.0  0.5  1.3  1.0  2.3   4.3  2023              [POR]  
1565  1.0  0.0  0.4  0.2  0.0   1.0  2023              [CHO]  
1566  1.7  0.7  0.0  0.0  1.7   8.0  2023          [IND/IND]  
1567  7.0  0.0  1.0  4.0  5.0  16.0  2023              [BRK]  
1568  7.0  3.0  0.0  2.0  3.0   3.0  2023              [MEM]  

[870 rows x 31 columns]

In [74]:
df_filtered.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp',
       'TOTAL TEAM'],
      dtype='object')

## Ahora creare una player_id para todos los jugadores teniendo en cuenta que hay jugadores que se  repiten. A estos hay que asignarle la misma id cada vez.

In [75]:
# Obtén los jugadores únicos
unique_players = df_filtered['Player'].unique()

# Crea un diccionario para mapear nombres de jugadores a IDs
player_id_map = {player: idx + 1 for idx, player in enumerate(unique_players)}

# Asigna IDs a los jugadores en el DataFrame
df_filtered['player_id'] = df_filtered['Player'].map(player_id_map)

C:\Users\Agus\AppData\Local\Temp\ipykernel_12692\495808539.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['player_id'] = df_filtered['Player'].map(player_id_map)


In [76]:
np.shape(df_filtered)

(870, 32)

In [77]:
df_filtered.isnull().sum()

Player        0
Pos           0
Age           0
Team          0
GP            0
GS            0
MP            0
FG            0
FGA           0
FG%           0
3P            0
3PA           0
3P%           0
2P            0
2PA           0
2P%           0
eFG%          0
FT            0
FTA           0
FT%           0
ORB           0
DRB           0
TRB           0
AST           0
STL           0
BLK           0
TOV           0
PF            0
PTS           0
Temp          0
TOTAL TEAM    0
player_id     0
dtype: int64

In [78]:
df_filtered.columns

Index(['Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
       '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
       'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp',
       'TOTAL TEAM', 'player_id'],
      dtype='object')

In [79]:
# Agrupamos por 'player_id' y verificamos si cada grupo tiene un solo nombre único en la columna 'Player'
validation = df_filtered.groupby('player_id')['Player'].nunique()

# Filtramos para encontrar cualquier 'player_id' que tenga más de un nombre asociado
invalid_ids = validation[validation > 1]

if invalid_ids.empty:
    print("¡Éxito! Todos los player_id tienen un único nombre de jugador asignado.")
else:
    print("Se encontraron inconsistencias en los siguientes player_id:")
    print(invalid_ids)

¡Éxito! Todos los player_id tienen un único nombre de jugador asignado.


In [80]:
df_filtered.isnull().sum()

Player        0
Pos           0
Age           0
Team          0
GP            0
GS            0
MP            0
FG            0
FGA           0
FG%           0
3P            0
3PA           0
3P%           0
2P            0
2PA           0
2P%           0
eFG%          0
FT            0
FTA           0
FT%           0
ORB           0
DRB           0
TRB           0
AST           0
STL           0
BLK           0
TOV           0
PF            0
PTS           0
Temp          0
TOTAL TEAM    0
player_id     0
dtype: int64

In [81]:
df_filtered.head()

Player Pos  Age   Team  GP  GS    MP   FG   FGA    FG%   3P  \
1        Jaylen Adams  PG   24  [MIL]   6   0   2.8  0.2   1.3  0.125  0.0   
3         Bam Adebayo   C   23  [MIA]  26  26  33.6  7.4  12.9  0.573  0.1   
5   Ty-Shon Alexander  SG   22  [PHO]   3   0   2.7  0.0   1.0  0.000  0.0   
8       Jarrett Allen   C   22  [TOT]  28  10  26.2  4.4   6.8  0.642  0.0   
11    Al-Farouq Aminu  PF   30  [ORL]   2   0   8.0  0.5   0.5  1.000  0.5   

    3PA   3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  AST  \
1   0.3  0.00  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5  0.3   
3   0.2  0.40  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2  5.3   
5   0.3  0.00  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3  0.3  0.3   
8   0.1  0.25  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1  9.0  1.6   
11  0.5  1.00  0.0   0.0  0.000  1.500  0.0  1.0  0.000  0.5  1.5  2.0  1.0   

    STL  BLK  TOV   PF   PTS  Temp         TOTAL TEAM  player_id  
1   0.0  0.0  0.0  0.2   0.3  2021              [MIL]          1  
3   1.0  1.0  3.0  2.6  19.9  2021      [MIA/MIA/MIA]          2  
5   0.0  0.0  0.0  0.3   0.0  2021              [PHO]          3  
8   0.5  1.6  1.5  1.6  12.3  2021  [BRK/CLE/CLE/CLE]          4  
11  0.0  0.0  1.0  0.0   1.5  2021              [ORL]          5

Se modifica el formato de las ids.

In [82]:
df_filtered['player_id'] = df_filtered['player_id'].astype('int')

C:\Users\Agus\AppData\Local\Temp\ipykernel_12692\377496901.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['player_id'] = df_filtered['player_id'].astype('int')


### Se crea una llave primaria compuesta unificada con la id de jugadores y la temporada en la que jugó.

Esto se decidió para que la tabla de hechos tenga todas ids unicas.

In [83]:
# Crear una nueva columna 'primary_key' que combine 'id' y 'Temp'
df_filtered['primary_key_composed'] = df_filtered['player_id'].astype(str) + "_" + df_filtered['Temp'].astype(str)

# Verificar las primeras filas de la tabla con la nueva columna
print(df_filtered[['player_id', 'Temp', 'primary_key_composed']].head())

    player_id  Temp primary_key_composed
1           1  2021               1_2021
3           2  2021               2_2021
5           3  2021               3_2021
8           4  2021               4_2021
11          5  2021               5_2021


C:\Users\Agus\AppData\Local\Temp\ipykernel_12692\1707180468.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_filtered['primary_key_composed'] = df_filtered['player_id'].astype(str) + "_" + df_filtered['Temp'].astype(str)


In [84]:
# Verificar si hay duplicados en la columna 'id'
duplicated_ids = df_filtered[df_filtered['primary_key_composed'].duplicated(keep=False)]

# Mostrar los duplicados encontrados (si existen)
if not duplicated_ids.empty:
    print("Se encontraron los siguientes duplicados en la columna 'primary_key_composed':")
    print(duplicated_ids)
else:
    print("No se encontraron duplicados en la columna 'primary_key_composed'.")



No se encontraron duplicados en la columna 'primary_key_composed'.


Verificamos el exito de nuestro codigo, observando los duplicados.

In [85]:
# Verificar si hay duplicados en la columna 'id'
duplicated_ids = df_filtered[df_filtered['primary_key_composed'].duplicated(keep=False)]

# Mostrar los duplicados encontrados (si existen)
if not duplicated_ids.empty:
    print("Se encontraron los siguientes duplicados en la columna 'primary_key_composed':")
    print(duplicated_ids)
else:
    print("No se encontraron duplicados en la columna 'primary_key_composed'.")


No se encontraron duplicados en la columna 'primary_key_composed'.


In [86]:
df_filtered

Player Pos  Age   Team  GP  GS    MP   FG   FGA    FG%   3P  \
1          Jaylen Adams  PG   24  [MIL]   6   0   2.8  0.2   1.3  0.125  0.0   
3           Bam Adebayo   C   23  [MIA]  26  26  33.6  7.4  12.9  0.573  0.1   
5     Ty-Shon Alexander  SG   22  [PHO]   3   0   2.7  0.0   1.0  0.000  0.0   
8         Jarrett Allen   C   22  [TOT]  28  10  26.2  4.4   6.8  0.642  0.0   
11      Al-Farouq Aminu  PF   30  [ORL]   2   0   8.0  0.5   0.5  1.000  0.5   
...                 ...  ..  ...    ...  ..  ..   ...  ...   ...    ...  ...   
1564      Justin Minaya  SF   23  [POR]   4   0  22.3  1.8   5.8  0.304  0.8   
1565       Kobi Simmons  SG   25  [CHO]   5   0   5.6  0.2   1.2  0.167  0.2   
1566          Gabe York  SG   29  [IND]   3   0  18.7  2.7   7.0  0.381  2.0   
1567       RaiQuan Gray  PF   23  [BRK]   1   0  35.0  6.0  12.0  0.500  2.0   
1568      Jacob Gilyard  PG   24  [MEM]   1   0  41.0  1.0   3.0  0.333  1.0   

      3PA    3P%   2P   2PA    2P%   eFG%   FT  FTA    FT%  ORB  DRB  TRB  \
1     0.3  0.000  0.2   1.0  0.167  0.125  0.0  0.0  0.000  0.0  0.5  0.5   
3     0.2  0.400  7.3  12.7  0.576  0.576  5.1  6.0  0.841  1.9  7.3  9.2   
5     0.3  0.000  0.0   0.7  0.000  0.000  0.0  0.0  0.000  0.0  0.3  0.3   
8     0.1  0.250  4.3   6.6  0.651  0.645  3.6  4.7  0.758  2.9  6.1  9.0   
11    0.5  1.000  0.0   0.0  0.000  1.500  0.0  1.0  0.000  0.5  1.5  2.0   
...   ...    ...  ...   ...    ...    ...  ...  ...    ...  ...  ...  ...   
1564  3.0  0.250  1.0   2.8  0.364  0.370  0.0  0.3  0.000  0.8  3.0  3.8   
1565  1.0  0.200  0.0   0.2  0.000  0.250  0.4  0.4  1.000  0.2  0.6  0.8   
1566  6.0  0.333  0.7   1.0  0.667  0.524  0.7  0.7  1.000  0.0  2.0  2.0   
1567  5.0  0.400  4.0   7.0  0.571  0.583  2.0  2.0  1.000  3.0  6.0  9.0   
1568  3.0  0.333  0.0   0.0  0.000  0.500  0.0  0.0  0.000  0.0  4.0  4.0   

      AST  STL  BLK  TOV   PF   PTS  Temp         TOTAL TEAM  player_id  \
1     0.3  0.0  0.0  0.0  0.2   0.3  2021              [MIL]          1   
3     5.3  1.0  1.0  3.0  2.6  19.9  2021      [MIA/MIA/MIA]          2   
5     0.3  0.0  0.0  0.0  0.3   0.0  2021              [PHO]          3   
8     1.6  0.5  1.6  1.5  1.6  12.3  2021  [BRK/CLE/CLE/CLE]          4   
11    1.0  0.0  0.0  1.0  0.0   1.5  2021              [ORL]          5   
...   ...  ...  ...  ...  ...   ...   ...                ...        ...   
1564  1.0  0.5  1.3  1.0  2.3   4.3  2023              [POR]        546   
1565  1.0  0.0  0.4  0.2  0.0   1.0  2023              [CHO]        547   
1566  1.7  0.7  0.0  0.0  1.7   8.0  2023          [IND/IND]        471   
1567  7.0  0.0  1.0  4.0  5.0  16.0  2023              [BRK]        548   
1568  7.0  3.0  0.0  2.0  3.0   3.0  2023              [MEM]        549   

     primary_key_composed  
1                  1_2021  
3                  2_2021  
5                  3_2021  
8                  4_2021  
11                 5_2021  
...                   ...  
1564             546_2023  
1565             547_2023  
1566             471_2023  
1567             548_2023  
1568             549_2023  

[870 rows x 33 columns]

Reordenamos las columnas en base a una estandarización propuesta por el equipo encargado de la arquitectura.

In [87]:
df_filtered = df_filtered[['primary_key_composed', 'player_id', 'Player', 'Pos', 'Age', 'Team', 'GP', 'GS', 'MP', 'FG', 'FGA', 'FG%',
            '3P', '3PA', '3P%', '2P', '2PA', '2P%', 'eFG%', 'FT', 'FTA', 'FT%',
            'ORB', 'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'Temp',
            'TOTAL TEAM']]

In [88]:
df_filtered = df_filtered.rename(columns={'Temp': 'season_id'})
df_filtered = df_filtered.rename(columns={'Team': 'team_abbreviation'})
df_filtered = df_filtered.rename(columns={'Player': 'player_name'})
df_filtered = df_filtered.rename(columns={'Pos': 'position'})
df_filtered = df_filtered.rename(columns={'Age': 'age'})

In [89]:
print(df_filtered.dtypes)

primary_key_composed     object
player_id                 int64
player_name              object
position                 object
age                       int64
team_abbreviation        object
GP                        int64
GS                        int64
MP                      float64
FG                      float64
FGA                     float64
FG%                     float64
3P                      float64
3PA                     float64
3P%                     float64
2P                      float64
2PA                     float64
2P%                     float64
eFG%                    float64
FT                      float64
FTA                     float64
FT%                     float64
ORB                     float64
DRB                     float64
TRB                     float64
AST                     float64
STL                     float64
BLK                     float64
TOV                     float64
PF                      float64
PTS                     float64
season_i

In [90]:
np.shape(df_filtered)

(870, 33)

Se guarda la tabla de hechos final.

In [91]:
df_filtered.to_csv('player_final_merged_with_stats.csv', index=False)